pkg/api/metadata.go

```diff
@@ -91,6 +92,35 @@ func splitLink(value string) (rel, link string) {
 	return
 }
 
+func UnmarshalMetadata(resp *http.Response, md *Metadata) {
+	// Resolve a URL against the request URL (same as `resp.Location()` just ignoring errors)
+	resolveURL := func(u string) string {
+		if resp.Request != nil && resp.Request.URL != nil {
+			if uu, err := resp.Request.URL.Parse(u); err == nil {
+				return uu.String()
+			}
+		}
+		return u
+	}
+
+	// For Link headers we only need to look between the brackets
+	linkURL := regexp.MustCompile("<[^>]+>")
```

Comment:
not that I expect this to be significant, but should we move the regex compilation out of this path and have it as a package level variable so we only compile it once?